torch/csrc/api/include/torch/data/init.h

```diff
@@ -0,0 +1,171 @@
+#pragma once
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <torch/data/datasets/chunk.h>
+namespace py = pybind11;
+
+namespace torch {
+namespace data {
+
+void init_dataset_bindings(PyObject* module);
+void init_dataset_bindings_impl(PyObject* module);
+void init_dataset_bindings_test(PyObject* module);
+
+namespace samplers {
+class SamplerWrapper : public Sampler<> {
+ public:
+  explicit SamplerWrapper(
+      std::shared_ptr<Sampler<>> sampler,
+      size_t total_stride = 0,
+      size_t current_stride = 0)
+      : sampler_(std::move(sampler)),
+        current_stride_(current_stride),
+        total_stride_(total_stride) {}
+  virtual ~SamplerWrapper() = default;
+
+  /// Resets the underlying sampler to a new set of indices.
+  void reset(torch::optional<size_t> new_size = nullopt) override {
+    sampler_->reset(new_size);
+  };
+
+  /// Sets the underlying sampler to a new stride
+  /// WARNING: `current_stride_` is not changed by the sampler `reset` routine
+  void set_current_stride(size_t stride) {
+    current_stride_ = stride;
+  };
+
+  /// Sets the total number of strides for the underlying sampler
+  /// WARNING: `total_stride_` is not changed by the sampler `reset` routine
+  void set_total_stride(size_t total) {
```

Comment:
Please change this to set_total_strides() instead of stride.

Also, regarding the warning. I would explain, the total strides and the current stride depend on the cluster setup and the number of workers used in each dataloader, so the reset will not change them.

Reply:
good idea.